046-analyse-broker-metrics-before-rolling-kafka.md

```diff
@@ -0,0 +1,93 @@
+# Analyse broker metrics before rolling Kafka
+
+This proposal describes a solution for KafkaRoller's shortcoming with brokers in log recovery.
+
+## Current situation
+
+The current logic in KafkaRoller:
+- Take a list of Kafka pods and reorder them based on the readiness.
+- Roll unready pods first without considering why it could be unready.
+- Before restarting a Kafka pod, consider if it is a controller or if it has an impact on the availability.
```

Comment:
maybe it's worth making clear these two points:
- controller is rolled as last pod.
- "impact on the availability" is related to avoid under replicated partitions for example.